app/models/bank_account.rb

```diff
@@ -24,4 +24,8 @@ def assign_unlinked_transactions
     end
     count
   end
+
+  def last_transaction_date
+    bank_transactions.order(date: :desc).first.try(:date)
+  end
```

Comment:
I'm a little hesitant, because it invites developers to use the method, and it can't be preloaded.
That said, it makes sense to put it here, so let's leave it.

Reply:
Also, I think we can use the `&.` operator, that in my view is more readable (and less lenient):
`bank_transactions.order(date: :desc).first&.date`